wmi-adapter/Tests/wmi.tests.ps1

```diff
@@ -52,10 +52,58 @@ Describe 'WMI adapter resource tests' {
         $LASTEXITCODE | Should -Be 0
         $r | Should -Not -BeNullOrEmpty
         $res = $r | ConvertFrom-Json
+
         $res.results[1].result.actualState.result[0].properties.Name | Should -Not -BeNullOrEmpty
         $res.results[1].result.actualState.result[0].properties.BootupState | Should -BeNullOrEmpty
         $res.results[1].result.actualState.result[1].properties.Caption | Should -Not -BeNullOrEmpty
         $res.results[1].result.actualState.result[1].properties.BuildNumber | Should -BeNullOrEmpty
-        $res.results[1].result.actualState.result[4].properties.AdapterType | Should -BeLike "Ethernet*"
+        $res.results[1].result.actualState.result[4].properties.Name | Should -Not -BeNullOrEmpty
+    }
+
+    It 'Set does not work without input for resource' -Skip:(!$IsWindows) {
+        $out = dsc resource set --resource root.cimv2/Win32_Environment --input '{}' 2>&1
+        $out[0] | Should -BeLike "*No valid properties found in the CIM class 'Win32_Environment' for the provided properties.*"
+    }
+
+    It 'Set does not work without a key property' -Skip:(!$IsWindows) {
+        $i = @{
+            VariableValue = "TestValue"
+            UserName = ("{0}\{1}" -f $env:USERDOMAIN, $env:USERNAME) # Read-only property is key, but we require a key property to be set
+        } | ConvertTo-Json
+        
+        $out = dsc resource set -r root.cimv2/Win32_Environment -i $i 2>&1
+        $out[0] | Should -BeLike "*All key properties in the CIM class 'Win32_Environment' are read-only, which is not supported.*"
```

Comment:
Since `UserName` is a key, the adapter needs it to find the right instance, so I would expect this to error that not all key properties are provided?